src/main.rs

```diff
@@ -45,26 +45,42 @@ fn build_poi_properties(row: &Row) -> Result<Vec<Property>, String> {
         .collect())
 }
 
-fn build_poi(row: Row, geofinder: &AdminGeoFinder) -> Poi {
+fn build_poi(row: Row, geofinder: &AdminGeoFinder) -> Option<Poi> {
     let name: String = row.get("name");
     let class: String = row.get("class");
-    let lat = row.get("lat");
-    let lon = row.get("lon");
-    let admins = geofinder.get(&geo::Coordinate { x: lat, y: lon });
-
-    Poi {
-        id: build_poi_id(&row),
-        coord: Coord::new(lat, lon),
-        administrative_regions: admins,
-        poi_type: PoiType {
-            id: class.clone(),
-            name: class,
-        },
-        label: name.clone(),
-        name: name,
-        weight: 0.,
-        zip_codes: vec![],
-        properties: build_poi_properties(&row).unwrap_or(vec![]),
+    let lat: Option<f64> = match row.get_opt("lat") {
```

Comment:
I think you can write this:
```rust
let lat = row.get_opt("lat")?
        .map_err(|e| warn!("impossible to get lat for {} because {}", name, e))
        .ok()?;
```
this way lat is an f64 and `?` early return a `None` (and we get a log on error)

the method becomes

```rust
fn build_poi(row: Row, geofinder: &AdminGeoFinder) -> Option<Poi> {
    let name: String = row.get("name");
    let class: String = row.get("class");
    let lat = row.get_opt("lat")?
        .map_err(|e| warn!("impossible to get lat for {} because {}", name, e))
        .ok()?;
    let lon = row.get_opt("lon")?
        .map_err(|e| warn!("impossible to get lon for {} because {}", name, e))
        .ok()?;
    let admins = geofinder.get(&geo::Coordinate { x: lat, y: lon });
    Some(Poi {
        id: build_poi_id(&row),
        coord: Coord::new(lat, lon),
        administrative_regions: admins,
        poi_type: PoiType {
            id: class.clone(),
            name: class,
        },
        label: name.clone(),
        name: name,
        weight: 0.,
        zip_codes: vec![],
        properties: build_poi_properties(&row).unwrap_or(vec![]),
    })
}
```

to be checked though

Reply:
Nice! It worked as expected.
An example of log obtained:
"impossible to get lat for Форт № 1 because type conversion error: a Postgres value was `NULL`"